### ISSUE TITLE
Jupyter Notebook Not Running

### ISSUE THREAD

**Issue by learning123456** [jupyter/notebook#4326]
I've been training a machine learning model running on Jupyter notebooks and the model was training fine before. I didn't change any of the code but wanted to retrain the model. The model started to train and print out the updated loss after each iteration (every ~10 seconds), but then stopped printing out the loss. The model was supposed to train a few hours ago but hasn't moved. The kernel says its still running, the cell still has the asterisk: everything seems to be working. I tried restarting everything, yet keep running into the same issue. It stops printing out the loss at different points in the training each time, and I know the code works as its run successfully many times before.

Is there anything I could try? Thanks.

**Comment by kevin-bates** [comment]
Some ideas...

- This is probably more a function of your configuration, but you might want to ensure the server on which your Notebook server is running has sufficient resources - especially if that server is hosting other Notebook servers (e.g., via JupyterHub) that are performing similar resource-intensive operations.

- If there's any kind of debug logging you can enable for your kernel (via the `argv` stanza in the kernelspec's kernel.json file), that may also yield some hints.

- When in this situation, try interrupting the cell's execution and ensure the interrupt is processed.  That will at least confirm that the kernel process is responding to interrupt signals.  This is a finer-grained approach to restart since restart terminates the kernel process altogether.  Depending on the kernel, the output produced from the successfully processed interrupt may lend some hints as to where things _may be blocked_ within your training.

- You could also try to locate the kernel process (assuming you have access to the Notebook server) and monitor its resource stats.  E.g., is memory or cpu time increasing despite no output appearing in the browser, etc.

**Comment by learning123456** [comment]
I tried interrupting the kernel this time; the interrupt is not processed and it says it's still running. 

The server definitely has enough resources, as I was previously successfully running the same model. I didn't change the code at all.

I checked the resource stats while it was running and saw that CPU usage, RAM usage, and GPU usage all dropped. 

Is there anything else I can try?
Thanks

**Comment by kevin-bates** [comment]
Thanks for the information.  So the kernel process is present yet interrupts are sent successfully, but not processed.  Process metrics have dropped (to zero?) implying a steady state.  Smells like things are blocked on something and _that something_ isn't cooperating.

Unfortunately, this will probably take a few more cycles so feel free to reach out to me via email and we can take this off-line.  We can update the issue later or others can chime in with their thoughts as well.

Just a few more ideas...
1. Is your training data the same as when this completed successfully?  If not, have you tried slicing the data into smaller chunks then build it back up?
2. You're probably using some third-party software (e.g., spark, tensorflow, etc.), have you enabled debug tracing via those applications/libraries?
3. Is the notebook log/output completely silent?  If you haven't already, I would start Notebook with `--log-level=DEBUG`.

**Comment by jyasonik** [comment]
I tried running the model in command prompt and it stopped running again in the middle of training. Apparently it's not an issue with the jupyter notebook, so i'll look at my code again. I really appreciate all the help.

**Comment by learning123456** [comment]
I tried training the model running in command prompt instead of in a jupyter notebook and the same thing happened. Apparently something's wrong with my code, not the notebook so I'll start looking down that route. I really appreciate all the help.
